.github/workflows/pull-request.yml

```
@@ -0,0 +1,1238 @@
+          else
+            echo "none=false" >> $GITHUB_OUTPUT
+          fi
+
+      # Diff-only i18n check to avoid existing legacy violations in untouched lines.
+      - name: Check for non-internationalized text (diff only)
+        if: steps.changed-src.outputs.none != 'true'
+        run: pnpm run check-i18n -- --diff --base ${{ github.event.pull_request.base.sha }} --head ${{ github.event.pull_request.head.sha }} ${{ steps.changed-src.outputs.files }}
+
+      - name: Check if the source and target branches are different
+        if: ${{ github.event.pull_request.base.ref == github.event.pull_request.head.ref }}
+        run: |
```

actionlint (1.7.9)

"github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)